images/prow-tests/Dockerfile

```diff
@@ -17,7 +17,7 @@ FROM debian:bullseye-20240211-slim AS base
 # Pinned versions of stuff we pull in
 ARG TARGETARCH
 ARG CLOUD_SDK_VERSION=432.0.0
-ARG KUBECTL_VERSION=v1.25.12
+ARG KUBECTL_VERSION=v1.28.7
```

Comment:
we could also go directly to 1.29.x, as the client is backwards compatible for some versions AFAIK

Reply:
happy with either option

Reply:
I updated it to 1.29.2, nice idea @creydr !